### ISSUE TITLE
Changing DatoCMS schema results in errors on gatsby develop

### ISSUE THREAD

**Issue by emik** [datocms/gatsby-source-datocms#35]
This can be fixed by changing the apiUrl to anything else, running `gatsby develop` then changing it back.

Many iterations of the error: `Calling "touchNode" with a nodeId is deprecated. Please pass an object containing a nodeId instead: touchNode({ nodeId: 'a-node-id' })` and

```
"touchNode" was called by gatsby-source-datocms
Calling "deleteNode" with a nodeId is deprecated. Please pass an object containing a full node instead: deleteNode({ node })
error UNHANDLED REJECTION


  TypeError: Cannot read property 'children' of undefined

  - actions.js:48 children.concat.children.map.child
    [theia]/[gatsby]/dist/redux/actions.js:48:39

  - Array.map

  - actions.js:47 findChildrenRecursively
    [theia]/[gatsby]/dist/redux/actions.js:47:42

  - actions.js:308 actions.deleteNode.args
    [theia]/[gatsby]/dist/redux/actions.js:308:29

  - bindActionCreators.js:7
    [theia]/[redux]/lib/bindActionCreators.js:7:35

  - api-runner-node.js:45 doubleBoundActionCreators.(anonymous function).args
    [theia]/[gatsby]/dist/utils/api-runner-node.js:45:13

  - gatsby-node.js:85
    [theia]/[gatsby-source-datocms]/gatsby-node.js:85:31

  - Array.forEach

  - gatsby-node.js:81 _callee$
    [theia]/[gatsby-source-datocms]/gatsby-node.js:81:38

  - gatsby-node.js:3 step
    [theia]/[gatsby-source-datocms]/gatsby-node.js:3:191

  - gatsby-node.js:3
    [theia]/[gatsby-source-datocms]/gatsby-node.js:3:361

  - next_tick.js:131 _combinedTickCallback
    internal/process/next_tick.js:131:7

  - next_tick.js:180 process._tickCallback
    internal/process/next_tick.js:180:9```

**Comment by k4roshi** [comment]
It can also be solved by deleting the .cache folder

**Comment by igghera** [comment]
+1 for @k4roshi 's solution

**Comment by ec8or** [comment]
This actually happens on some data changes as well, if I upload a new image and save while my project is running it crashes and I have to delete cache and restart.

**Comment by ryancoughlin** [comment]
Running into the same issue. Thoughts on things to try? Been trying to trouble shoot with no luck.

**Comment by szanella** [comment]
We're also experiencing this issue. Deleting the .cache folder works, but in my opinion it's not a viable workflow in the long run. Any updates on what causes the problem?

**Comment by jakegibson** [comment]
Same issue, looks like the fragment generation in onPreExtractQueries in gatsby-node does not get run on DatoCMS change.  I haven't gotten a chance to dig into it more.  Has anyone found a workaround?

**Comment by stefanoverna** [comment]
Solved in version `2.1.0-beta.1`, soon to be released! 

https://github.com/datocms/gatsby-source-datocms/blob/2.1.0/CHANGELOG.md#v210
